### PR TITLE
Basis for chunk upload with APIV2, DI for testing and refactored ios-core.

### PR DESCRIPTION
**Pull request by adrien-coye**

### CODE REVIEW

File: kDriveCore/Utils/Logging.swift [hunk 1]
```diff
@@ -71,7 +71,7 @@ public enum Logging {
     private static func initNetworkLogging() {
         #if DEBUG
             if !Bundle.main.isExtension {
-                Atlantis.start(hostName: ProcessInfo.processInfo.environment["hostname"])
+                Atlantis.start(hostName: "adrien-coye-mbp.local.")
```
Comment (adrien-coye) [inline kDriveCore/Utils/Logging.swift:74]: TODO: Rollback

File: kDrive/UI/Controller/Home/HomeViewController.swift [hunk 2]
```diff
@@ -192,6 +196,55 @@ class HomeViewController: UICollectionViewController, SwitchDriveDelegate, Switc
...
+        let button = UIButton(type: .roundedRect)
+        button.frame = CGRect(x: 0, y: 50, width: 120, height: 40)
+        button.setTitle("Button", for: .normal)
+        button.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
+        view.addSubview(button)
+        self.button = button
+    }
+
+    // TODO: clean
+    @objc func buttonAction() {
```
Comment (adrien-coye) [inline kDrive/UI/Controller/Home/HomeViewController.swift:210]: TODO: remove

File: kDriveCore/Data/Models/Upload/UploadChunk.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+
+public struct UploadChunk: Decodable {
```
Comment (adrien-coye) [inline kDriveCore/Data/Models/Upload/UploadChunk.swift:21]: Still needed ?

File: kDriveCore/Data/UploadQueue/BackgroundUploadSessionManager.swift [hunk 4]
```diff
@@ -20,6 +20,7 @@ import CocoaLumberjackSwift
 import Foundation
 import Sentry
 
+// check for existing url session switcheroo
```
Comment (adrien-coye) [inline kDriveCore/Data/UploadQueue/BackgroundUploadSessionManager.swift:23]: TODO: check

File: kDriveCore/Data/_MoveToCore/ApiFetcher+Dispatch.swift [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        
+        let defaultSession = URLSession(configuration: .default)
+        
+        var urlRequest = URLRequest(url: endpoint)
+        urlRequest.httpMethod = requestable.method.rawValue
+        
+        // append body
+        switch requestable.body {
+        case .requestBody(let data):
+            urlRequest.httpBody = data // TODO missing data header and footer ?
```
Comment (adrien-coye) [inline kDriveCore/Data/_MoveToCore/ApiFetcher+Dispatch.swift:114]: TODO

File: kDriveCore/Data/_MoveToCore/ApiFetcher+Dispatch.swift [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        case .POSTParameters(let parameters):
+            urlRequest.httpBody = Data() // TODO encode fields
```
Comment (adrien-coye) [inline kDriveCore/Data/_MoveToCore/ApiFetcher+Dispatch.swift:116]: TODO

File: kDriveCore/Data/_MoveToCore/ApiFetcher+Dispatch.swift [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        case .none:
+            break
+        }
+        
+        let tuple = try await defaultSession.data(for: urlRequest)
+        print("data: \(tuple.0)\n response: \(tuple.1) \n")
+        
+        fatalError()
+        
+//        return (tuple.0, tuple.1)
```
Comment (adrien-coye) [inline kDriveCore/Data/_MoveToCore/ApiFetcher+Dispatch.swift:126]: clean

File: kDriveTests/kDriveCore/DriveApiTests.swift [hunk 8]
```diff
@@ -30,6 +30,8 @@ class FakeTokenDelegate: RefreshTokenDelegate {
     func didFailRefreshToken(_ token: ApiToken) {}
 }
 
+// FIXME: Missing env
```
Comment (adrien-coye) [inline kDriveTests/kDriveCore/DriveApiTests.swift:33]: FIXME

File: kDriveTests/kDriveCore/DriveFileManagerTests.swift [hunk 9]
```diff
@@ -25,6 +25,8 @@ import XCTest
 
 @testable import kDrive
 
+// FIXME: missing ENV
```
Comment (adrien-coye) [inline kDriveTests/kDriveCore/DriveFileManagerTests.swift:28]: FIXME

File: Project.swift [hunk 10]
```diff
@@ -86,9 +87,12 @@ let project = Project(name: "kDrive",
                                  platform: .iOS,
                                  product: .unitTests,
                                  bundleId: "com.infomaniak.drive.tests",
-                                 deploymentTarget: .iOS(targetVersion: "13.0", devices: [.iphone]),
+                                 deploymentTarget: .iOS(targetVersion: "13.4", devices: [.iphone]),
                                  infoPlist: .file(path: "kDriveTests/Tests.plist"),
                                  sources: "kDriveTests/**",
+                                 resources: [
+                                     "kDriveTests/**/*.jpg"
```
Comment (adrien-coye) [inline Project.swift:94]: I used tuist to import the ressourses I need in the test target. cc @PhilippeWeidmann

File: kDrive/UI/Controller/Menu/SwitchUserViewController.swift [hunk 11]
```diff
@@ -45,7 +45,7 @@ class SwitchUserViewController: UIViewController {
                         _ = try await self.accountManager.updateUser(for: account, registerToken: false)
                     }
                 }
-                try await group.waitForAll()
+                try await group.waitForAll() // This wait in an async task ?
```
Comment (adrien-coye) [inline kDrive/UI/Controller/Menu/SwitchUserViewController.swift:48]: TODO: check

Comment (adrien-coye) [reply kDrive/UI/Controller/Menu/SwitchUserViewController.swift:48]: I'd like to check this with you @PhilippeWeidmann

File: kDriveCore/Data/_MoveToCore/ApiFetcher+Dispatch.swift [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+            return try await dispatchAlamofire(requestable)
+        case .NSURLSession:
+            return try await dispatchNSURLSession(requestable, inBackground: false)
+        case .NSURLSessionBackground:
+            return try await dispatchNSURLSession(requestable, inBackground: true)
+        }
+    }
+    
+    // TODO: Update existing method in core with encoding
+    func __authenticatedRequest(_ endpoint: Endpoint,
```
Comment (adrien-coye) [inline kDriveCore/Data/_MoveToCore/ApiFetcher+Dispatch.swift:40]: TODO: move to core.

File: kDriveCore/Data/Models/Upload/UploadLiveSession.swift [hunk 13]
```diff
@@ -0,0 +1,38 @@
+//
```
Comment (PhilippeWeidmann) [inline kDriveCore/Data/Models/Upload/UploadLiveSession.swift:1]: Xcode adds a comment for each new file

File: kDriveCore/Data/Models/Upload/UploadSession.swift [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+
+public struct UploadSession: Decodable {
+    public var directoryID: UInt64?
```
Comment (PhilippeWeidmann) [inline kDriveCore/Data/Models/Upload/UploadSession.swift:22]: Use `directoryId` (and others `fileId`, `driveId`)

File: kDriveShareExtension/ShareNavigationViewController.swift [hunk 15]
```diff
@@ -17,21 +17,20 @@
...
-import InfomaniakCore
+import InfomaniakCoreUI
 import InfomaniakLogin
 import kDriveCore
+import InfomaniakDI
 
 class ShareNavigationViewController: TitleSizeAdjustingNavigationController {
 
-    private var accountManager: AccountManager!
+    @InjectService var accountManager: AccountManageable
```
Comment (PhilippeWeidmann) [inline kDriveShareExtension/ShareNavigationViewController.swift:27]: We are in an extension so we need to setup DI from scratch (Same for FileProviderExtension)

File: kDrive/AppDelegate+BGAppRefresh.swift [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+
+import Foundation
+import BackgroundTasks
+import kDriveCore
+import CocoaLumberjackSwift
+import InfomaniakDI
+
+extension AppDelegate {
+    
+    var bgScheduler :BGTaskScheduler {
```
[the remark this answers is not in the document]
Comment (adrien-coye) [reply kDrive/AppDelegate+BGAppRefresh.swift:27]: Renamed `backgroundTaskScheduler`. Furthermore, now using DI.

File: kDriveShareExtension/ShareNavigationViewController.swift (hunk 15 above)
Comment (adrien-coye) [reply kDriveShareExtension/ShareNavigationViewController.swift:27]: I moved the `FactoryService` code to kDriveCore, so I could use it from all the extensions.

We will probably need a mechanism that allows for setting up Factories in kDrive _and_ kDriveCore, but for now this is enough.

**Review by PhilippeWeidmann — APPROVED** [review]

**Pull request merged**